structured_data/2021_02_26_scikit_learn_pipelines/scikit_learn_pipelines_and_lightbm_titanic_dataset.ipynb

```diff
@@ -0,0 +1,1157 @@
+              "                                                                   OrdinalEncoder())]),\n",
+              "                                                  ['Sex', 'Embarked'])])),\n",
+              "                ('estimator',\n",
+              "                 LGBMClassifier(n_estimators=20, random_state=42))])"
+            ]
+          },
+          "metadata": {
+            "tags": []
+          },
+          "execution_count": 26
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Ri8b2LzoEybg",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "43886fe7-49ab-4088-c8f7-133086c73fab"
+      },
+      "source": [
+        "model = full_pipe.fit(X_train, y_train)\r\n",
+        "\r\n",
+        "print(f'Train accuracy: {model.score(X_train, y_train):.2f}')\r\n",
+        "print(f'Test accuracy: {model.score(X_test, y_test):.2f}')"
+      ],
+      "execution_count": 27,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Train accuracy: 0.85\n",
+            "Test accuracy: 0.79\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "uQFysEh0NiQJ"
+      },
+      "source": [
+        "Not the performance gain you may have hoped for, as the dataset is too 'simple'. But not to worry, you will encounter more challenging situations in the field. And when you do, now you know how easy it is to get started on the latest Gradient Boosting Trees using scikit learn best practices.\r\n",
+        "\r\n",
+        "For the interested reader, we have added a section on hyper parameter tuning on the pipeline object. Make sure to check it out below!\r\n",
+        "\r\n",
+        "\r\n",
+        "\r\n",
+        "# In summary\r\n",
+        "\r\n",
+        "You have been introduced to the `scikit-learn` [`Pipeline`](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html) object. This object chains preprocessing and modeling together for easy execution and hyper parameter tuning (see the bonus content). In addition, you have seen how you can easily combine scikit-learn with the latest GBMs to kickstart your next project."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RP99aRcMn96j"
+      },
+      "source": [
+        "---\r\n",
+        "\r\n",
+        "## Bonus Content\r\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "E0cKk-uKMaAj"
+      },
+      "source": [
+        "\r\n",
+        "### 5. Hyper parameter tuning\r\n",
+        "\r\n",
+        "Just building a model isn't all, you also need to find the right parameters for the model! This is important especially for gradient boosting as they are very sensitive to hyper parameters. \r\n",
```

Comment:
> as they are very sensitive to hyper parameters.

Can we add a source here?

Reply:
Would something like this be sufficient: https://bradleyboehmke.github.io/HOML/gbm.html#tuning-strategy ?

Reply:
Refers to this paper: https://arxiv.org/pdf/1802.09596.pdf figure two. But the link to the quote might be more digestable.

Reply:
Added the high level source